src/gui/static/src/app/services/wallet.service.ts

```diff
@@ -88,13 +83,30 @@ export class WalletService {
       .flatMap(addresses => this.apiService.get('outputs', {addrs: addresses}));
   }
 
+  allPendingTransactions(): Observable<any> {
+    return this.apiService.get('pendingTxs');
+  }
+
   pendingTransactions(): Observable<any> {
     return this.pendingTxs.asObservable();
   }
 
   refreshPendingTransactions() {
-    this.apiService.get('pendingTxs').subscribe(txs => {
-      this.pendingTxs.next(txs);
+    this.wallets.first().subscribe(wallets => {
+      Observable.forkJoin(wallets.map(wallet => this.apiService.get('wallet/transactions', { id: wallet.filename })))
+        .subscribe(pending => {
+          this.pendingTxs.next(
+            pending
+              .filter(response => response.transactions.length > 0)
+              .map(response => response.transactions)
+              .reduce((txs, tx) => {
+                if (!txs.find(t => t.transaction.txid === tx.transaction.txid)) {
```

Comment:
Throwing "ERROR TypeError: Cannot read property 'txid' of undefined" both with real transactions and with simulated responses.

For initial testing, you can force the node to always return one transaction per wallet by adding this code on the line 520 of src/gui/wallet.go:

```
testTx := visor.UnconfirmedTxn {}
txns = make([]visor.UnconfirmedTxn, 1)
txns[0] = testTx;
```